mypyc/genops.py

```diff
@@ -2011,6 +2011,43 @@ def visit_op_expr(self, expr: OpExpr) -> Value:
             return self.shortcircuit_expr(expr)
         return self.binary_op(self.accept(expr.left), self.accept(expr.right), expr.op, expr.line)
 
+    def translate_eq_cmp(self,
+                         lreg: Value,
+                         rreg: Value,
+                         expr_op: str,
+                         line: int) -> Optional[Value]:
+        ltype = lreg.type
+        rtype = rreg.type
+        if not (isinstance(ltype, RInstance) and ltype == rtype):
+            return None
+
+        class_ir = ltype.class_ir
+        # Check whether any subclasses of the operand redefines __eq__.
+        cmp_varies_at_runtime = False
+        for cmp_op in ('__eq__', '__ne__'):
+            method = class_ir.get_method(cmp_op)
+            if any(subc.get_method(cmp_op) is not method
```

Comment:
I think it is time for ClassIR to grow a method that queries whether a method is overridden by any subclasses, which can then be used here, in the emitfunc method call optimization, and instead of has_bool.

Reply:
Added it